app/models/piece.rb

```diff
@@ -45,115 +40,128 @@ def move_to!(new_x, new_y)
     end
   end
 
-  def is_obstructed?(x1, y1, x2, y2)
-    vertical_move = x1 === x2
-    if vertical_move
-      if (y1 < y2)
-        small_y = y1
-        big_y = y2
+  def is_vertically_obstructed?(new_x, new_y)
+    x1 = self.x_position
+    y1 = self.y_position
+    if (y1 < new_y)
+      small_y = y1
+      big_y = new_y
+    else
+      small_y = new_y
+      big_y = y1
+    end
+    i = small_y + 1
+
+    while i < big_y  do
+      if ($board[x1][i] === nil)
+        i +=1
       else
-        small_y = y2
-        big_y = y1
+        return false
       end
+    end
+    return true
+  end
 
-      i = small_y + 1
+  def is_horizontally_obstructed?(new_x, new_y)
+    x1 = self.x_position
+    y1 = self.y_position
+    if (x1 < new_x)
+      small_x = x1
+      big_x = new_x
+    else
+      small_x = new_x
+      big_x = x1
+    end
 
-      while i < big_y  do
-        if ($board[x1][i] === nil)
-          i +=1
+    i = small_x + 1
+
+    while i < big_x do
+      if ($board[i][y1] === nil)
+        i +=1
+      else
+        return false
+      end
+    end
+    return true
+  end
+
+  def is_diagonally_obstructed(new_x, new_y)
+    x1 = self.x_position
+    y1 = self.y_position
+    if (x1 - new_x).abs != (y1 - new_y).abs
+      raise "Invalid move"
+    end
+    
+    if ((x1 < new_x) && (y1 < new_y)) 
+      x = x1 + 1
+      y = y1 + 1
+      while x < new_x do
+        if ($board[x][y] === nil)
+          x += 1
+          y += 1
         else
           return false
         end
       end
       return true
     end
-    
-    horizontal_move = y1 === y2
-    if horizontal_move
-      if (x1 < x2)
-        small_x = x1
-        big_x = x2
-      else
-        small_x = x2
-        big_x = x1
-      end
-
-      i = small_x + 1
 
-      while i < big_x do
-        if ($board[i][y1] === nil)
-          i +=1
+    if ((x1 < new_x) && (y1 > new_y)) 
+      x = x1 + 1
+      y = y1 - 1
+      while x < new_x do
+        if ($board[x][y] === nil)
+          x += 1
+          y -= 1
         else
           return false
         end
       end
       return true
     end
-    
-    
-    diagonal_move = (x1 != x2) && (y1 != y2)
-    if diagonal_move
-      if (x1 - x2).abs != (y1 - y2).abs
-        raise "Invalid move"
-      end
-      
-      if ((x1 < x2) && (y1 < y2)) 
-        x = x1 + 1
-        y = y1 + 1
-        while x < x2 do
-          if ($board[x][y] === nil)
-            x += 1
-            y += 1
-          else
-            return false
-          end
-        end
-        return true
-      end
 
-      if ((x1 < x2) && (y1 > y2)) 
-        x = x1 + 1
-        y = y1 - 1
-        while x < x2 do
-          if ($board[x][y] === nil)
-            x += 1
-            y -= 1
-          else
-            return false
-          end
+    if ((x1 > new_x) && (y1 < new_y)) 
+      x = x1 - 1
+      y = y1 + 1
+      while x > new_x do
+        if ($board[x][y] === nil)
```

Comment:
@Nikkovichert there are still a few references to $board[x][y].. What you are referring to is our game and checking if there are pieces on the x/y cell, correct? so it should be sth like `game.pieces.find_by(x_position: x, y_position: y)`